pkg/pillar/attest/attest.go

```diff
@@ -0,0 +1,346 @@
+	case EventInternalQuoteRecvd:
+		return "EventInternalQuoteRecvd"
+	case EventRetryTimerExpiry:
+		return "EventRetryTimerExpiry"
+	case EventNonceMismatch:
+		return "EventNonceMismatch"
+	case EventQuoteMismatch:
+		return "EventQuoteMismatch"
+	case EventNoQuoteCertRecvd:
+		return "EventNoQuoteCertRecvd"
+	case EventAttestEscrowRecorded:
+		return "EventAttestEscrowRecorded"
+	case EventAttestEscrowFailed:
+		return "EventAttestEscrowFailed"
+	case EventAttestSuccessful:
+		return "EventAttestSuccessful"
+	case EventRestart:
+		return "EventRestart"
+	default:
+		return "Unknown Event"
+	}
+}
+
+//String returns human readable string of a State
+func (state State) String() string {
+	switch state {
+	case StateNone:
+		return "StateNone"
+	case StateNonceWait:
+		return "StateNonceWait"
+	case StateInternalQuoteWait:
+		return "StateInternalQuoteWait"
+	case StateAttestWait:
+		return "StateAttestWait"
+	case StateAttestEscrowWait:
+		return "StateAttestEscrowWait"
+	case StateRestartWait:
+		return "StateRestartWait"
+	case StateComplete:
+		return "StateComplete"
+	default:
+		return "Unknown State"
+	}
+}
+
+//Verifier needs to be implemented by the consumer of this package
+//It contains interface definitions for interacting with attestation server
+type Verifier interface {
+	SendNonceRequest(ctx *Context) error
+	SendAttestQuote(ctx *Context) error
+	SendAttestEscrow(ctx *Context) error
+}
+
+//TpmAgent needs to be implemented by the consumer of this package
+//It contains interface definitions for interacting with TPM manager
+type TpmAgent interface {
+	SendInternalQuoteRequest(ctx *Context) error
+}
+
+//Various error codes to be returned to this package from external interfaces
+var (
+	ErrControllerUnavailable = errors.New("Controller is unavailable")
+	ErrNonceMismatch         = errors.New("Nonce Mismatch")
+	ErrQuoteMismatch         = errors.New("Quote Mismatch")
+	ErrNoCertYet             = errors.New("No Cert Found")
+	ErrInfoTokenInvalid      = errors.New("Provided Integrity Token is Invalid")
+	ErrTpmAgentUnavailable   = errors.New("TPM agent is unavailable")
+)
+
+//External interfaces to the state machine
+//For unit-testing, these will be redirected to their mock versions.
+var tpmAgent TpmAgent
+var verifier Verifier
+
+//Context has all the runtime context required to run this state machine
+type Context struct {
+	event                 Event
+	state                 State
+	restartTimer          *time.Timer
+	eventTrigger          chan Event
+	retryTime             time.Duration
+	restartRequestPending bool
+}
+
+//Transition represents an event triggered from a state
+type Transition struct {
+	event Event
+	state State
+}
+
+//EventHandler represents a handler function for a Transition
+type EventHandler func(*Context) error
+
+//the state machine
+var transitions = map[Transition]EventHandler{
+	{EventInitialize, StateNone}:                       handleInitializeWhlNone,                       //goes to NonceWait
```

Comment:
"At<state>" might be more obvious naming scheme than "Whl<state>"

Reply:
> "At" might be more obvious naming scheme than "Whl"

Done.